application-forum-ui/src/main/resources/ForumCode/RecentForumDiscussionsMacro.xml

```diff
@@ -0,0 +1,418 @@
+      <id>
+        <disabled>0</disabled>
+        <name>id</name>
+        <number>1</number>
+        <prettyName>Macro id</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <code>{{velocity}}
+#set ($discard = $xwiki.ssx.use('ForumCode.RecentForumDiscussionsMacro'))
+#set ($limit = $xcontext.macro.params.limit)
+#if (!$limit)
+  #set ($limit = 5)
+#end
+## get from all the forums most recent N topics and topics' answers
+#set ($query = ", BaseObject as topic, BaseObject as answer where doc.fullName in (topic.name, answer.name) and doc.fullName not in ('ForumCode.TopicTemplate', 'ForumCode.AnswerTemplate') and topic.className='ForumCode.TopicClass' and answer.className='ForumCode.AnswerClass' order by doc.date desc")
+#set ($results = $services.query.hql($query).addFilter('unique').setLimit($limit).execute())
+(% class='recentForumDiscussions' %)
+#foreach ($result in $results)
+  ## @action param is used at the end to build the translation key, which reflects the action performed by an user
+  ## since answers don't have a title, the label for their URL is the title of the parent topic
+  #set ($document = $xwiki.getDocument($result))
+  #if ($document.getObject('ForumCode.TopicClass'))
+    ## Topics
+    #set ($topicTitle = $document.title)
+    #if ($document.version == '1.1')
+      #set ($action = 'createdTopic')
+    #else
+      #set ($action = 'editedTopic')
+    #end
+  #else
+    ## Answers 
+    #set ($topicTitle = $xwiki.getDocument($document.parent).title)
```

Comment:
Not a good idea to keep using the (deprecated) parent property exclusively. You can choose to ignore this now, but will probably come back to this when it will become a problem (like being removed from platform). 

If you have the time, it would be better to handle it in a cleaner way.